src/pydata_sphinx_theme/__init__.py

```diff
@@ -40,11 +40,12 @@ def update_config(app):
             )
 
     # Validate icon links
-    if not isinstance(theme_options.get("icon_links", []), list):
-        raise ExtensionError(
-            "`icon_links` must be a list of dictionaries, you provided "
-            f"type {type(theme_options.get('icon_links'))}."
-        )
+    if theme_options.get("icon_links") is not None:
+        if not isinstance(theme_options.get("icon_links", []), list):
```

Comment:
I can see why on current `main, setting `icon_links = None` in your `conf.py` would cause an error here. But if you don't want icon links, why not just leave `icon_links` undefined?  Genuinely curious if there's a reason you need this.